plugins/inputs/logfile/fileconfig.go

```diff
@@ -203,3 +251,55 @@ func (config *FileConfig) isMultilineStart(logValue string) bool {
 	}
 	return config.MultiLineStartPatternP.MatchString(logValue)
 }
+
+func (config *FileConfig) shouldFilterLog(event logs.LogEvent) bool {
+	if config.Filters == nil || len(config.Filters) < 1 {
+		return false
+	}
+	config.sampleCount += 1
+	start := time.Now()
```

Comment:
Optional / suggestion - you could do:
```
start := time.Now()
defer func() {
    elapsed := time.Since(start)
    if config.sampleCount >= sampleThreshold {
        log.Printf("foo")
        config.sampleCount = 0
    }
}()
```
Then remove it from the 3 places below.

Reply:
ahhh that's a good point. I can make that change.

Reply:
As of right now I'm dropping these debug logs. Will keep this in mind if I reintroduce it though